src/store/api/worldUsers.ts

```diff
@@ -96,46 +96,22 @@ export const worldUsersApi = createApi({
           .collection("users")
           .where("enteredVenueIds", "array-contains-any", relatedLocationIds);
 
-        const unsubscribeListener = worldUsersQuery.onSnapshot((snapshot) => {
-          // TODO: implement this properly + cleanup this placeholder stuff
-          const added = snapshot
-            .docChanges()
-            .filter((docChange) => docChange.type === "added").length;
+        let hasLoadedDataInitially = false;
 
-          const removed = snapshot
-            .docChanges()
-            .filter((docChange) => docChange.type === "removed").length;
+        const unsubscribeListener = worldUsersQuery.onSnapshot((snapshot) => {
+          const snapshotSize = snapshot.size;
 
-          const modified = snapshot
-            .docChanges()
-            .filter((docChange) => docChange.type === "modified").length;
+          // NOTE: Don't delay the update of the first load.
+          if (snapshotSize > 1 && !hasLoadedDataInitially) {
```

Comment:
Why > 1 here? Should it be > 0?

Reply:
> Why > 1 here?
It's because of how Firebase sends the data. The first snapshot always contains 1 item, which is your own profile, which you most probably fetch when you get the auth data.

> Should it be > 0?
If I do this, then the next batch which is `all the other users` will be delayed